src/plugins/data_views/public/services/has_data.ts

```diff
@@ -77,6 +82,41 @@ export class HasData {
     return source;
   };
 
+  private getIndicesViaSearch = async ({
+    http,
+    pattern,
+    showAllIndices,
+  }: {
+    http: HttpStart;
+    pattern: string;
+    showAllIndices: boolean;
+  }): Promise<boolean> =>
+    http
+      .post<IndicesViaSearchResponse>(`/internal/search/ese`, {
```

Comment:
This is risky because this uses async search and we might get partial result here (which could be empty) 
This either should use `data.search` service or at least hit `/internal/search/es` which would use sync search

Reply:
we can't use data.search because of cyclical dependency. We are doing the 2nd part right?